### ISSUE TITLE
Payin via PayPal

### ISSUE THREAD

**Issue by Changaco** [liberapay/liberapay.com#424]
MangoPay has implemented PayPal payins, this feature is currently in private beta (source: https://github.com/Mangopay/mangopay2-python-sdk/pull/64).

**Comment by volth** [comment]
Any progress here?

**Comment by Changaco** [comment]
@volth No. I think PayPal payins are still in private beta, which we haven't applied for because this feature isn't a priority for us. Do you need PayPal payins as soon as possible for some reason?

**Comment by volth** [comment]
I am getting questions from donors who prefer paypal. Maybe it would increase the donations, maybe not.

**Comment by Changaco** [comment]
+1 from https://social.coop/@dixongexpat/187780

**Comment by Changaco** [comment]
I reached out to MangoPay but the answer I got is that PayPal payins are in fact not available and that they're only working payouts for now. 🙄

**Comment by volth** [comment]
Any chance to work with PayPal directly, without MangoPay?

**Comment by Changaco** [comment]
It's difficult for us to do things without MangoPay because they're the ones who actually hold our users' money (and have the licence to do so).

Also PayPal is quite temperamental and I suspect that Liberapay alone doesn't have enough weight (transaction volume) to establish a balanced relationship with them.

But I'll look into it.

**Comment by Changaco** [comment]
According to MangoPay we can't add PayPal ourselves.

I've researched the issue myself and at this point I won't say that it's impossible, but it's definitely not easy. Firstly, what we want to do may not fit PayPal's policies. Secondly, low volume = high fees on PayPal, unless we obtain the nonprofit discount, but since the collected money wouldn't be for us we may not get the discount. Finally there is the issue of transferring the money to the recipient MangoPay wallets.

Braintree (a MangoPay competitor owned by PayPal and used by Gratipay) doesn't operate in the EU yet, and we can't run a donations platform from the US for the same reasons Gratipay can't.

**Comment by Changaco** [comment]
Support for PayPal payins has been added again to the python module: <https://github.com/Mangopay/mangopay2-python-sdk/pull/120>. Strange for a feature that supposedly doesn't exist, isn't it?

**Comment by Changaco** [comment]
+1 from https://mamot.fr/@pintoch/4540563

**Comment by ericfont** [comment]
Just adding my +1 vote, and subscribing to this issue...once I get notification that it is implemented, I will become a donator.

**Comment by artsyhonker** [comment]
I think this would make it easier, conceptually, for a lot of potential donors. People seem to feel safer using PayPal to pay in than they feel about giving their bank details to a new-to-them website. To avoid fees eating too much of the money, one solution might be a minimum pay-in amount if paying in via PayPal: maybe €100 or thereabouts? (I realise this is pointless if MangoPay won't allow it.)

**Comment by Changaco** [comment]
It looks like [Adaptive Payments](https://developer.paypal.com/docs/classic/products/adaptive-payments/) could allow us to support PayPal in parallel to Mangopay (instead of PayPal through Mangopay).

**Comment by Changaco** [comment]
Closing in favour of #1187.
